modules/core/test/test_math.cpp

```diff
@@ -2618,6 +2618,34 @@ TEST(Core_SolveCubic, regression_27323)
     }
 }
 
+TEST(Core_SolveCubic, regression_27748)
+{
+    // a is extremely small relative to others (approx 1.8e-19 ratio),
+    // causing instability in standard cubic formula.
+    double a = 1.56041e-17;
+    double b = 84.4504;
+    double c = -96.795;
+    double d = 13.6826;
+
+    Mat coeffs = (Mat_<double>(1, 4) << a, b, c, d);
+    Mat roots;
+
+    // We expect the solver to detect the negligible 'a' and solve as quadratic:
+    // 84.4504x^2 - 96.795x + 13.6826 = 0
+    // Discriminant ~ 4745.8
+    // Roots approx: 0.156, 0.990
+
+    int n = solveCubic(coeffs, roots);
+
+    EXPECT_GE(n, 2); // Should find at least 2 real roots (as quadratic)
+
+    for(int i = 0; i < n; i++)
+    {
+        double r = roots.at<double>(i);
+        EXPECT_NEAR(r, 0.5, 1.0);
```

Comment:
It means that roots are validated in +-1 range. It's too much. See documentation: https://google.github.io/googletest/reference/assertions.html#EXPECT_NEAR